kartothek/core/common_metadata.py

```diff
@@ -547,8 +547,10 @@ def _remove_diff_header(diff):
 
 def _diff_schemas(first, second):
     # see https://issues.apache.org/jira/browse/ARROW-4176
-    first_pyarrow_info, _ = str(first).split("metadata\n--------")
-    second_pyarrow_info, _ = str(second).split("metadata\n--------")
+
+    # Pyarrow >0.16.0 do not include the metadata anymore
+    first_pyarrow_info = str(first).split("metadata\n--------")[0]
+    second_pyarrow_info = str(second).split("metadata\n--------")[0]
```

Comment:
This isn't set in stone yet.

Reply:
The code here should at least not raise anymore. 
I could offer https://github.com/JDASoftwareGroup/kartothek/pull/247 to not break arrow integration due to minor changes like these format changes

Reply:
Yeah, I question whether a str/repr should be regarded as stable from commit to commit. This output is designed for humans, not machines, at least that is how it has been developed so far. Would it be better to check only the data contained in the schema (field names, types, metadata keys/values)?

Reply:
We only process this diff because the output with metadata was way to verbose and the diff was essentially useless. This function strips the metadata and offers a better human readable diff. I now realize that we should probably just diff `schema.remove_metadata()` which should achieve the same but should be much more stable.

I'll take another look at #247 because I do agree that we shouldn't commit to the repr as a public API.

That is/was the intention of https://issues.apache.org/jira/browse/ARROW-4176 (I'll link this comment for more context)